### ISSUE TITLE
Jquery deferred exception - Uncaught TypeError

### ISSUE THREAD

**Issue by MarkLannen** [Montana-Media-Arts/441-WebTech-Spring2019#17]
Any help would be appreciated! I've linked my repo and copied my html and js below my question.

I'm not able to get my image(s) to load via this command:

$(imageArray[0].theImageSelector).attr("src", imageArray[0].theImagePath);

I am seeing the attached error in the console. I've tried googling the errors but haven't been able to understand the answers enough to fix my issue. 

![Screen Shot 2019-03-09 at 7 39 05 PM](https://user-images.githubusercontent.com/31551434/54079996-6ea6b680-42a4-11e9-8ddc-5ab010206509.png)

my repo is here: https://github.com/MarkLannenUM/MART441_Work/tree/master/HW-8

Below I've copied html and js.

Thank!

<!DOCTYPE html>
<html>

    <head>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
        <script src="scripts/script.js"></script>
        <link href="styles/style.css" type="text/css" rel="stylesheet">
    
    </head>

    <body onload="initializeArray();">
        
        <br>
        <button id="move">Show Images</button>
        <br>
        <img src="img/cloud-1.jpg" id="image-Id">
    </body>

</html>

// Image class
var imageArray = new Array();

class Images {
    constructor(selector, imagePath) {
        this.selector = selector;
        this.imagePath = imagePath;
    }

    get theImageSelector() {
        return this.selector;
    }

    get theImagePath() {
        return this.imagePath;
    }
}

function initializeArray() {
    var image1 = new Images("#image-Id", "img/cloud-1.jpg");
    var image2 = new Images("#image-Id", "img/milkyWay-1.jpg");
    var image3 = new Images("#image-Id", "img/nebula-1.jpg");
    var image4 = new Images("#image-Id", "img/universe-1.jpg");

    imageArray.push(image1);
    imageArray.push(image2);
    imageArray.push(image3);
    imageArray.push(image4);
}

console.log(imageArray);
// Jquery

$(document).ready(function() {
    $(imageArray[0].theImageSelector).attr("src", imageArray[0].theImagePath);
    $("button").click(function(){        
        $(imageArray[0]).fadeOut().fadeIn();
    });

});

**Comment by michaelcassens** [comment]
Hi Mark,

On line 39, you just need to add .theSelector to your array and then it
should work;

$(imageArray[0].theImageSelector).fadeOut().fadeIn();

Hope this helps!

Michael

**Comment by MarkLannen** [comment]
Thanks - but I'm afraid that did not do it.  :(  

Something is telling me that there is a gap in the logic of how I'm accessing src, but I just can't work it out. 

Do the attached error messages from the console mean anything to you?


![Screen Shot 2019-03-09 at 7 39 05 PM](https://user-images.githubusercontent.com/31551434/54080242-b62f4180-42a8-11e9-9aae-e0f25d0c4436.png)

**Comment by MarkLannen** [comment]
Line 37 is this line:

$(imageArray[0].theImageSelector).attr("src", imageArray[0].theImagePath);

**Comment by michaelcassens** [comment]
Hmm.. I had just grabbed your repo and updated that line.  Should I look
elsewhere?

Michael

**Comment by MarkLannen** [comment]
Did it work for you?

**Comment by MarkLannen** [comment]
I did make the change you suggested but I'm still getting those errors and the image won't load. 

Confused...
:)

**Comment by michaelcassens** [comment]
Yes, however, I noticed in the code you sent directly in your email that
some of your code was outside of the script tag which would result in
variables being undefined.

Hope this helps!

Michael

**Comment by MarkLannen** [comment]
Sorry, but now I'm really confused. I'm using an external js file linked in the html. I don't understand how my variables could be outside the script tag...

And I don't understand why it would work for you but not me? I've opened it in atom and vs code just in case there was something wrong there, but I'm getting the same errors with both.

**Comment by MarkLannen** [comment]
I've debugged it to this point - line 36 in my js file.

$(imageArray[2].theImageSelector).attr("src", "img/cloud-1.jpg");

When I hard coded the img id it worked, but when I tried to use **imageArray[2].theImageSelector**
it failed. I guess I know where the problem is occurring, I just don't know what it is or how to fix it.

My repo has been updated: https://github.com/MarkLannenUM/MART441_Work/tree/master/HW-8

Any pointers?

Thanks!

**Comment by michaelcassens** [comment]
Hi Mark,

I will take a look as soo as I can.

Michael

**Comment by michaelcassens** [comment]
Hi Mark,

Would you mind giving me more specifics on when it fails?

Thanks,

Michael

**Comment by MarkLannen** [comment]
On a whim when I woke up this morning I tried opening the html file directly in Chrome and it worked with no errors in the console. I then tried viewing it in live server from vs Code (which is how I've been working this semester) and it didn't work and I saw the same errors in the console.  

Then I switched default browser in live server settings to Firefox (from Chrome) and it worked!

The problem must be opening live server in Chrome.

So I think I'll stop using live server for the time being until I can figure out what is causing the problem. This actually explains a few odd behaviours I've seen this semester. :)

Anyway - thanks for your help, and sorry to ask you to help fix something that didn't actually need fixing!
